### ISSUE TITLE
Resource Spike For query/AccountTxs 

### ISSUE THREAD

**Issue by andrewnguyen22** [pokt-network/pocket-core#1090]
**Describe the bug**
```
I run an under powered non-validator on mainnet as a canary.  It does not provide or request relays.  Just a plain ole full node.
it is a standard Digitial Ocean $10/month, 1 vcpu, 2 gig of memory box.
I can reliably and repeatedly kill it with a single query command which will take it from a stable 226 mb of memory use.  to 1869 mb instantly at it will die shortly thereafter (next block for sure, sometimes instantly)
```
Before query
![Screenshot from 2020-10-05 12-29-22](https://user-images.githubusercontent.com/18633790/95112153-983cf380-070e-11eb-941a-af17b88be47a.png)
After query
![Screenshot from 2020-10-05 12-33-46](https://user-images.githubusercontent.com/18633790/95112109-85c2ba00-070e-11eb-87a2-13d314ced333.png)

NOTE: `The process kswapd0 is the process that manages virtual memory`
Observe Pocket Memory Spike from a single accountTxs Query

This did not return the result.

This sometimes results in:
`E[2020-10-05|16:53:25.292] data has been corrupted in last height of consensus WAL module=consensus err="DataCorruptionError[failed to read data: EOF (read: 15, wanted: 257)]" height=6566`

Might be related to #1089

**Comment by andrewnguyen22** [comment]
Just reproduced this issue with RC-0.5.0
Before Query: *Memory 61 MB*
<img width="1442" alt="Screen Shot 2020-10-05 at 3 11 55 PM" src="https://user-images.githubusercontent.com/18633790/95121708-6aab7680-071d-11eb-9f1d-1efca458da64.png">
After Query: *Memory 3.4GB*
<img width="1469" alt="Screen Shot 2020-10-05 at 3 13 00 PM" src="https://user-images.githubusercontent.com/18633790/95121720-6f702a80-071d-11eb-82ac-5ca1c2c540d5.png">


```
http://localhost:8081/v1/query/accounttxs
the http status code was not okay: 400, and the status was: 400 Bad Request, 
with a response of {"code":400,"message":"TxSearch: Post http://localhost:26657: EOF"}
```

**Comment by andrewnguyen22** [comment]
Tendermint 33.7 Did not fix this issue:
<img width="1473" alt="Screen Shot 2020-10-05 at 3 22 10 PM" src="https://user-images.githubusercontent.com/18633790/95122524-a561de80-071e-11eb-98e9-4855796d91b7.png">

**Comment by andrewnguyen22** [comment]
Since Pocket-Core's side of this seems pretty limited:
```
tmClient := app.GetClient()
	defer func() { _ = tmClient.Stop() }()
	_, err = hex.DecodeString(addr)
	if err != nil {
		return nil, err
	}
	query := fmt.Sprintf(messageSenderQuery, addr)
	page, perPage = checkPagination(page, perPage)
	res, err = tmClient.TxSearch(query, prove, page, perPage, "asc")
	return
```
Turning my attention/investigation to Tendermint:
```
func TxSearch(ctx *rpctypes.Context, query string, prove bool, page, perPage int, orderBy string) (
```

**Comment by andrewnguyen22** [comment]
Upon first inspection, it seems like Tendermint executes the query without limits and applies the pagination afterward:
```
func TxSearch(ctx *rpctypes.Context, query string, prove bool, page, perPage int) (*ctypes.ResultTxSearch, error) {
	// if index is disabled, return error
	if _, ok := txIndexer.(*null.TxIndex); ok {
		return nil, fmt.Errorf("Transaction indexing is disabled")
	}

	q, err := tmquery.New(query)
	if err != nil {
		return nil, err
	}

	results, err := txIndexer.Search(q)
	if err != nil {
		return nil, err
	}

	totalCount := len(results)
	perPage = validatePerPage(perPage)
	page, err = validatePage(page, perPage, totalCount)
```

**Comment by andrewnguyen22** [comment]
Opened separate issue to tackle pagination: https://github.com/pokt-network/pocket-core/issues/1100

**Comment by andrewnguyen22** [comment]
So it seems like the txSearch option is rather inefficient.

The indexer gets the keys and values for all transactions that satisfy the query
`results, err := env.TxIndexer.Search(ctx.Context(), q)`
Sorts the transactions in some order 
`// sort results (must be done before pagination)`
Then paginates it:
```
apiResults := make([]*ctypes.ResultTx, 0, pageSize)
	for i := skipCount; i < skipCount+pageSize; i++ {
		r := results[i]

		var proof types.TxProof
		if prove {
			block := env.BlockStore.LoadBlock(r.Height)
			proof = block.Data.Txs.Proof(int(r.Index)) // XXX: overflow on 32-bit machines
		}

		apiResults = append(apiResults, &ctypes.ResultTx{
			Hash:     r.Tx.Hash(),
			Height:   r.Height,
			Index:    r.Index,
			TxResult: r.Result,
			Tx:       r.Tx,
			Proof:    proof,
		})
	}
```

All in memory
